File: src/domain/entities/operational/SessionReference.ts
/**
 * Canonical identifier for a hosted WhatsApp session.
 */
export class SessionReference {
  constructor(
    public readonly provider: string,
    public readonly workspaceId: number,
    public readonly sessionId: string,
  ) {}

  public toKey(): string {
    return `${this.provider}:${this.workspaceId}:${this.sessionId}`;
  }

  public toLogLabel(): string {
    return `${this.provider} session ${this.sessionId} (WS: ${this.workspaceId})`;
  }

  public belongsTo(providerId: string): boolean {
    return this.provider === providerId;
  }
}

File: sdks/java/src/main/java/com/digows/whatsappgateway/operational/SessionReference.java
package com.digows.whatsappgateway.operational;

public record SessionReference(
  String provider,
  int workspaceId,
  String sessionId
)
{
  public SessionReference
  {
    if (provider == null || provider.isBlank())
    {
      throw new IllegalArgumentException("SessionReference requires a non-empty provider.");
    }

    if (workspaceId <= 0)
    {
      throw new IllegalArgumentException("SessionReference requires a positive workspaceId.");
    }

    if (sessionId == null || sessionId.isBlank())
    {
      throw new IllegalArgumentException("SessionReference requires a non-empty sessionId.");
    }
  }

  public String toKey()
  {
    return provider + ":" + workspaceId + ":" + sessionId;
  }

  public String toLogLabel()
  {
    return provider + " session " + sessionId + " (WS: " + workspaceId + ")";
  }
}
